src/rest/AnalyticsEndpoint.ts

```diff
@@ -52,7 +52,7 @@ export class AnalyticsEndpoint {
       return this.DEFAULT_ANALYTICS_URI;
     }
 
-    const [basePlatform] = endpoint.options.restUri.split('/rest');
+    const [basePlatform] = endpoint.options.restUri.replace('platform', 'analytics').split('/rest');
```

Comment:
Perhaps it would be a good idea to have a more restrictive replacement? E.g.:
```suggestion
    const [basePlatform] = endpoint.options.restUri
      .replace(/^(https?:\/\/)platform/, '$1analytics')
      .split('/rest');
```